src/dev/encoder.cpp

```diff
@@ -0,0 +1,141 @@
+#include <EVT/dev/encoder.hpp>
+
+namespace EVT::core::DEV {
+
+Encoder::Encoder(IO::GPIO& a, IO::GPIO& b, uint64_t range, uint64_t initialPosition, bool rollOver)
+    : a(a), b(b), range(range), position(initialPosition), rollOver(rollOver) {
+    if (position > range) {
+        position = range;
+    }
+    //setting instance variables
+    currentRelPos = readPinValues();
+    interruptChange = 0;
```

Comment:
This and the two below can be set by default in the `.hpp` instead of here. Putting them there just makes this method harder to mess up in the future